lib/tools/system-calls.js

```diff
@@ -835,14 +831,30 @@ systemCallMethods.waitForDevice = async function waitForDevice (appDeviceReadyTi
  * @throws {Error} If the device failed to reboot and number of retries is exceeded.
  */
 systemCallMethods.reboot = async function reboot (retries = DEFAULT_ADB_REBOOT_RETRIES) {
-  await this.shell(['stop'], {
-    privileged: true,
-  });
-  await B.delay(2000); // let the emu finish stopping;
-  await this.setDeviceProperty('sys.boot_completed', 0);
-  await this.shell(['start'], {
-    privileged: true,
-  });
+  try {
+    // Get root access so we can run the following shell commands that require root access
+    const {wasAlreadyRooted} = await this.root();
+
+    // Stop and re-start the device
+    await this.shell(['stop']);
+    await B.delay(2000); // let the emu finish stopping;
+    await this.setDeviceProperty('sys.boot_completed', 0, false);
+    await this.shell(['start']);
+
+    // Return root state to what it was before
+    if (!wasAlreadyRooted) {
+      await this.unroot();
+    }
+  } catch (e) {
```

Comment:
^ @mykola-mokhnach I changed reboot logic to not be 'privileged' because it was causing a root+unroot on the 3 different shell calls. Opted for just rooting before the calls and then unrooting after.